frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js

```diff
@@ -0,0 +1,405 @@
+        this.setState({ rows: [...copiedRows] });
+    };
+
+    changeSchemaName = (e) => {
+        if (!this.props.editPermission) return;
+        this.setState({ cur_schema_name: e.target.value });
+    };
+
+    moveFieldRow = (idx, direction) => () => {
+        if (!this.props.editPermission) return;
+        // -1 for moving up, 1 for moving down
+        const { rows } = this.state;
+        let copiedRows = [...rows];
+        let newIndex = idx + direction;
+        [copiedRows[newIndex], copiedRows[idx]] = [copiedRows[idx], copiedRows[newIndex]];
+        if (copiedRows[idx].from_schema_name !== this.props.item.schema_name) {
+            // if the last row we switched with was generated by addschema
+            // we should check and keep switching
+            // until we meet a non-addschema row or top/end of table
+            idx += direction;
+            newIndex += direction;
+            while (
+                newIndex >= 0 &&
+                newIndex < rows.length &&
+                rows[newIndex].from_schema_name !== this.props.item.schema_name
+            ) {
+                [copiedRows[newIndex], copiedRows[idx]] = [copiedRows[idx], copiedRows[newIndex]];
+                idx += direction;
+                newIndex += direction;
+            }
+        }
+        this.setState({ rows: copiedRows });
+    };
+
+    removeFieldRow = (idx) => () => {
+        if (!this.props.editPermission) return;
+        const { rows } = this.state;
+        let copiedRows = [...rows];
+        copiedRows.splice(idx, 1);
+        this.setState({ rows: copiedRows });
+    };
+
+    componentDidUpdate(prevProps, prevState) {
+        if (
+            prevProps.item.field_rows !== this.props.item.field_rows ||
+            prevProps.item.schema_name !== this.props.item.schema_name
+        ) {
+            this.setState({
+                rows: [...this.props.item.field_rows],
+                cur_schema_name: this.props.item.schema_name,
+            });
+        }
+        if (this.state.newAddedRow !== -1 && this.state.rows.length === prevState.rows.length + 1) {
+            document.getElementById('textbox-' + this.state.newAddedRow + '-0').focus();
+        }
+    }
+
+    render() {
+        const { classes, editPermission, focused, item } = this.props;
+        const { showSchemaDetail, rows } = this.state;
+        const schemaItem = item;
+        const schemaHeaders = schemaItem.header;
+        const schema_name = schemaItem.schema_name;
+        let headerHtml, bodyRowsHtml;
+        const explanations = {
+            field: 'Name of the field. ',
+            'generalized-path':
+                "Either a field of bundle's metadata (like uuid, name) or a file path inside the bundle prefixed by / (like '/stdout').\n",
+            'post-processor':
+                'Optional, a function that transforms the string value result of generalized-path, for example uuid uuid [0:8] will keep the first 8 digits of the uuid. ',
```

Comment:
"How to render the value (e.g., %.3f renders 3 decimal points, [0:8] takes the first 8 characters, duration renders seconds, size renders bytes)."